library/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/obfuscator/rules/MaskAllObfuscationRule.kt

```diff
@@ -0,0 +1,33 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay.internal.recorder.obfuscator.rules
+
+import android.widget.TextView
+import com.datadog.android.sessionreplay.internal.recorder.obfuscator.DefaultStringObfuscator
+import com.datadog.android.sessionreplay.internal.recorder.obfuscator.FixedLengthStringObfuscator
+
+internal class MaskAllObfuscationRule(
+    private val defaultStringObfuscator: DefaultStringObfuscator =
+        DefaultStringObfuscator(),
+    private val fixedLengthStringObfuscator: FixedLengthStringObfuscator =
+        FixedLengthStringObfuscator(),
+    private val textTypeResolver: TextTypeResolver =
+        TextTypeResolver(),
+    private val textValueResolver: TextValueResolver = TextValueResolver()
+) : TextValueObfuscationRule {
+    override fun resolveObfuscatedValue(textView: TextView): String {
```

Comment:
question/ How would it scale to obfuscating text in other elements (e.g. options in value pickers or dropdown lists)? Will we keep adding more methods to `TextValueObfuscationRule` interface? That could indicate very strong coupling and a need for [shotgun surgery](https://refactoring.guru/smells/shotgun-surgery) to change behaviour of a single component.

Reply:
By applying the strategy I explained above using the `parent` my plan is to always stick to these rules and only have 4 text types. The `parent` property will be used to check if the current `TextView` is part of a `SELECTOR` meaning it will be an `OPTION_TEXT`

Reply:
The only thing that I need to add later will be probably a `HideObfuscationRule` for fine grained but will see when I get there.